src/components/fields/AutoSave.js

```diff
@@ -1,37 +1,23 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { FormSpy } from 'react-final-form';
-import { isEqual } from 'lodash';
 
 // Coming from official example of async on blur saving
 // https://codesandbox.io/s/7k742qpo36
 
 // Mixed with the following
 // https://codesandbox.io/s/5w4yrpyo7k?from-embed
 
-class AutoSave extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = { values: props.values, submitting: false };
-    }
+const AutoSave = ({ values, save, debounce }) => {
+    useEffect(() => {
+        const timeout = setTimeout(() => save(values), debounce);
 
-    componentWillReceiveProps(nextProps) {
```

Comment:
Sorry, I had to rewrite this component because of this componentWillReceiveProps, which is deprecated in React